posthog/models/event/events_query.py

```diff
@@ -0,0 +1,173 @@
+import json
+from datetime import timedelta
+from typing import List, Optional, Tuple
+
+from dateutil.parser import isoparse
+from django.utils.timezone import now
+
+from posthog.api.element import ElementSerializer
+from posthog.api.utils import get_pk_or_uuid
+from posthog.clickhouse.client.connection import Workload
+from posthog.hogql import ast
+from posthog.hogql.constants import SELECT_STAR_FROM_EVENTS_FIELDS
+from posthog.hogql.parser import parse_expr, parse_order_expr
+from posthog.hogql.property import action_to_expr, has_aggregation, property_to_expr
+from posthog.hogql.query import execute_hogql_query
+from posthog.models import Action, Person, Team
+from posthog.models.element import chain_to_elements
+from posthog.schema import EventsQuery, EventsQueryResponse
+from posthog.utils import relative_date_parse
+
+QUERY_DEFAULT_LIMIT = 100
+QUERY_DEFAULT_EXPORT_LIMIT = 3_500
+QUERY_MAXIMUM_LIMIT = 100_000
+
+
+def run_events_query(
+    team: Team,
+    query: EventsQuery,
+) -> EventsQueryResponse:
+    # Note: This code is inefficient and problematic, see https://github.com/PostHog/posthog/issues/13485 for details.
+    # To isolate its impact from rest of the queries its queries are run on different nodes as part of "offline" workloads.
```

Comment:
I _think_ the online/offline split was reverted after a recent incident. Whether that needs to affect this comment is 🤷

Reply:
I think it was reverted too, but left the comment in place because the first line is still true.